### PR TITLE
bpo-26187: Tests for sqlite3 trace callback

### PR DESCRIPTION
**Pull request by AnishShah**

### CODE REVIEW

File: Lib/sqlite3/test/hooks.py [hunk 1]
```diff
@@ -248,6 +248,23 @@ def trace(statement):
                         "Unicode data %s garbled in trace callback: %s"
                         % (ascii(unicode_value), ', '.join(map(ascii, traced_statements))))
 
+    def CheckTraceCallbackContent(self):
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/hooks.py:251]: We should skip the test if sqlite3_prepare_v2 is not available.

Comment (berkerpeksag) [reply Lib/sqlite3/test/hooks.py:251]: The test name doesn't explain what we actually testing here.

File: Lib/sqlite3/test/hooks.py [hunk 2]
```diff
@@ -248,6 +248,23 @@ def trace(statement):
...
+        """
+        Test that the statement are correct. Fix for bpo-26187
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/hooks.py:253]: Docstring is not needed.

File: Lib/sqlite3/test/hooks.py [hunk 3]
```diff
@@ -248,6 +248,23 @@ def trace(statement):
...
+        traced_statements = []
+        def trace(statement):
+            traced_statements.append(statement)
+        con.set_trace_callback(trace)
+        queries = ["create table foo(x)", "insert into foo(x) values(1)"]
+        for query in queries:
+            con.execute(query)
+            con.commit()
+        queries.insert(1, "BEGIN ")
+        queries.append("COMMIT")
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/hooks.py:265]: These two lines can be removed. We only want to test that set_trace_callback doesn't produce any duplicate entries.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by palaviv** [comment]
This test don't fail before the fix in #359. I opened another PR (#461) with fixed test.

**Comment by berkerpeksag** [comment]
Thanks, Aviv. Next time please send your feedback as a comment and give the PR author some time to address your comments. Otherwise, we will end up with duplicate PRs.

Also note that we always locally run tests and make sure that the test fails before the fix is applied so the chance of merging a broken test is minimal.

**Comment by palaviv** [comment]
Sorry about that @berkerpeksag. I actually started by writing a comment on this PR but found out that what I do is to write the all test again. I am still having hard time understanding what @AnishShah did here as I wrote a test case in the issue.

**Pull request closed without merging**